### PR TITLE
docs: update readme

### PR DESCRIPTION
**Pull request by jplanckeel**
## Describe

Update readme docs.

### CODE REVIEW

**Review by gforien — APPROVED** [review]

**Review by Eriwyr — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,8 +4,26 @@ This project is a Prometheus exporter written in Golang, its goal is to provide
 
 The project includes an exporter that calculates the metric to provide according to the annotations defined on the hpa. and an API that allows to declare an event and a capacity multiplier
```
Comment (taufort) [inline README.md:5]:
```suggestion
The exporter exposes a metric calculated according to the annotations defined on the HPA. The project also exposes an API that allows to register an event with a capacity multiplier.
```

File: README.md [hunk 2]
```diff
@@ -4,8 +4,26 @@ This project is a Prometheus exporter written in Golang, its goal is to provide
...
 
+## Requirements
 
-## Annotations to add on HPA
+   - Prometheus Stack or Victoria Metrics Stack
+   - [Prometheus Adapater](https://github.com/kubernetes-sigs/prometheus-adapter) 
+
+*Info : It is quite possible to use this solution with another observability stack than Prometheus. For example Datadog or Newrelic, but we don't have a configuration example.*
```
Comment (taufort) [inline README.md:12]:
```suggestion
*Info: It is quite possible to use this solution with another observability stack than Prometheus. For example, Datadog or Newrelic, but we do not provide a configuration example.* 
```

File: README.md [hunk 3]
```diff
@@ -4,8 +4,26 @@ This project is a Prometheus exporter written in Golang, its goal is to provide
...
+## Kubernetes Deployement
+
+- Clone repo
+- Run this command with Helm3
+
+```bash
+helm install prescaling-exporter ./helm/prescaling-exporter -n prescaling-exporter --create-namespace
+```
+
+*You can use skaffold if you want.*
```
Comment (taufort) [inline README.md:24]: For a note, I'd rather use `>`:
```suggestion
> You can use skaffold if you want.
```

File: README.md [hunk 4]
```diff
@@ -4,8 +4,26 @@ This project is a Prometheus exporter written in Golang, its goal is to provide
...
+- Clone repo
+- Run this command with Helm3
+
+```bash
+helm install prescaling-exporter ./helm/prescaling-exporter -n prescaling-exporter --create-namespace
+```
+
+*You can use skaffold if you want.* 
+
+## Configure a Horizontal Pod Autoscaler
```
Comment (taufort) [inline README.md:26]:
```suggestion
## Configure an Horizontal Pod Autoscaler
```

File: README.md [hunk 5]
```diff
@@ -4,8 +4,26 @@ This project is a Prometheus exporter written in Golang, its goal is to provide
...
 
 To be able to pre-scale an application every day before a traffic spike, the only thing to do is to add the
```
Comment (taufort) [inline README.md:28]:
```suggestion
To be able to pre-scale an application every day before a traffic spike, you must add the
```

File: README.md [hunk 6]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+  - type: External
+    external:
+      metricName: "prescale_metric"
+      metricSelector:
+          matchLabels:
+            deployment: "{{ .Release.Name }}"
+      targetValue: 10
+```
+
+*it's important to set the target to 10, the value provided for scale is 11, the addition of pod will be gradual. Currently, the increment is carried out 10% by 10%.*
```
Comment (taufort) [inline README.md:64]:
```suggestion
> It's important to set the `targetValue` to 10. The metric's value provided by the exporter in order to scale is 11. The scale up of pods will be gradual. Currently, the increment is carried out 10% by 10%.
```

File: README.md [hunk 7]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+
+## Configure prometheus adapter
+
+A configuration example using the prometheus adapter to supply the metric to the Kubernetes cluster.
```
Comment (taufort) [inline README.md:68]:
```suggestion
Here is a configuration example using the prometheus adapter to supply the metric to the Kubernetes cluster:
```

File: README.md [hunk 8]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+    "resources":
+      "overrides":
+        "namespace":
+          "resource": "namespace"
+    "seriesQuery": "prescale_metric"
+```
+
+## Configure parameters 
+
+You can modify the application's settings to use different annotations or a different port. Configurable variable environments in the chart values are used to do this.
```
Comment (taufort) [inline README.md:83]:
```suggestion
You can modify the application's settings to use different annotations or a different port. Configurable variable environments in the chart values are used to do this:
```

File: README.md [hunk 9]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+
+Parameters            | Default values                            | Comment
+---                   | ---                                       | --- 
+Namespace             | "prescaling-exporter"                     | Namespace for PrescalingEvent
```
Comment (taufort) [inline README.md:87]: Why not set the namespace to `prescaling` by default? `exporter` is tightly coupled to Prometheus IMHO.

```suggestion
Namespace             | "prescaling"                     | Namespace for the prescaling stack
```

File: README.md [hunk 10]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+Port                  | "9101"                                    | Application port
+AnnotationEndTime     | "annotations.scaling.exporter.time.end"   | Annotation end in HPA for create metrique
+AnnotationStartTime   | "annotations.scaling.exporter.time.start" | Annotation start in HPA for create metrique
+AnnotationMinReplicas | "annotations.scaling.exporter.replica.min"| Annotation in min HPA for create metrique
```
Comment (taufort) [inline README.md:91]: In English, if you use "for + verb", you must always add "ing" to the verb. Ex: for creating metric.

Anyway, I suggest:
```suggestion
AnnotationEndTime     | "annotations.scaling.exporter.time.end"   | Prescaling end time
AnnotationStartTime   | "annotations.scaling.exporter.time.start" | Prescaling start time
AnnotationMinReplicas | "annotations.scaling.exporter.replica.min"| Minimum of desired replicas during the prescaling
```

File: README.md [hunk 11]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+LabelProject          | "project"                                 | label k8s where to retrieve the value for the project label of the metric
```
Comment (taufort) [inline README.md:92]:
```suggestion
LabelProject          | "project"                                 | k8s label used to add a label in the prescaling metric
```

File: README.md [hunk 12]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+# Prescaling Web server
+## OpenAPI docs
+
+In the application is provided a swagger which is accessible with the endpoint. 
+
+/swagger/index.html
+
+## Event API 
+
+To allow the platform to scale up and down on different schedules and with a multiplier, pre-scaling events can be requested by the DRBs or the API.
```
Comment (taufort) [inline README.md:103]:
```suggestion
To allow the platform to scale up and down on different schedules and with a multiplier, prescaling events can be registered by the DRBs or the API.
```

DRBs?

File: README.md [hunk 13]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+AnnotationStartTime   | "annotations.scaling.exporter.time.start" | Annotation start in HPA for create metrique
+AnnotationMinReplicas | "annotations.scaling.exporter.replica.min"| Annotation in min HPA for create metrique
+LabelProject          | "project"                                 | label k8s where to retrieve the value for the project label of the metric
+
+# Prescaling Web server
+## OpenAPI docs
+
+In the application is provided a swagger which is accessible with the endpoint. 
+
+/swagger/index.html
```
Comment (taufort) [inline README.md:99]:
```suggestion
This application provides a swagger UI which is accessible on  `/swagger/index.html`.
```

File: README.md [hunk 14]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+
+## Event API 
+
+To allow the platform to scale up and down on different schedules and with a multiplier, pre-scaling events can be requested by the DRBs or the API.
+
+The API allows the creation/modification and deletion of Prescaling Event CRDs in the cluster
+
+/api/v1/events/
```
Comment (taufort) [inline README.md:107]:
```suggestion
The following API allows the creation, modification and deletion of Prescaling Event CRDs in the cluster: `/api/v1/events/`.
```

File: README.md [hunk 15]
```diff
@@ -17,6 +35,84 @@ annotations.scaling.exporter.time.end | "hh:mm:ss"
...
+
+## Metrics
+
+The metrics are exposed with the endpoint
+
+/metrics
```
Comment (taufort) [inline README.md:113]:
```suggestion
The metrics are exposed on `/metrics` endpoint.
```

**Pull request merged**